SpatialGDK/Source/SpatialGDK/Public/SpatialView/CommandRequest.h

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Templates/UniquePtr.h"
+#include <improbable/c_worker.h>
+#include <improbable/c_schema.h>
+
+namespace SpatialGDK
+{
+
+struct CommandRequestDeleter
+{
+	void operator()(Schema_CommandRequest* CommandRequest) const noexcept
+	{
+		if (CommandRequest)
+		{
+			Schema_DestroyCommandRequest(CommandRequest);
+		}
+	}
+};
+
+using OwningCommandRequestPtr = TUniquePtr<Schema_CommandRequest, CommandRequestDeleter>;
+
+// An RAII wrapper for component data.
+class CommandRequest
+{
+public:
+	// Creates a new component data.
+	explicit CommandRequest(Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
```

Comment:
Perhaps my understanding is off here but I thought explicit is useful for single argument constructors?

Reply:
It means you can't do = {stuff, more stuff, something else}
I tend to only like allowing that if the stuff in the initializer list is actually the value of the thing being constructed.

Reply:
Ahh fair. 

"is actually the value of the thing being constructed." Sorry didn't fully understand this.
Does this mean that all arguments are used to directly initialize members?
 : ComponentId(InComponentId)

Reply:
Just coming back to this; will discuss offline.